python/tests/operations/add.py

```diff
@@ -292,6 +300,37 @@ def test_self_add_error(self):
             str(cm.exception),
             "B should be a TensorMap or a scalar value. ",
         )
+        self.assertTrue(equistore.equal(A, A_copy))  # check not modified in place
```

Comment:
I'm not sure we need to check every where that nothing have been modified, this is already checked in the two test functions above